docker_cli/image.py

```diff
@@ -0,0 +1,458 @@
+        str, optional
+            The output of the process, if `stdout` == PIPE.
+
+        Raises
+        -------
+        CalledProcessError
+            If the docker inspect command fails.
+        CommandNotFoundOnImageError:
+            When a command is attempted that is not recognized on the image.
+        """
+        cmd = ["docker", "run", f"--network={network}", "--rm"]
+        if interactive:
+            cmd += ["-i"]
+            if stdin.isatty():
+                cmd += ["--tty"]
+        cmd += [self._id, "bash"]
+        cmd += ["-ci"] if interactive else ["-c"]
+        cmd += split(f"'{command}'")
+
+        try:
+            result = run(
+                cmd,
+                text=True,
+                stdout=stdout,  # type: ignore
+                stderr=stderr,  # type: ignore
+                check=True
+            )
+            retval = result.stdout
+        except CalledProcessError as err:
+            if err.returncode == 127:
+                raise CommandNotFoundOnImageError(err, split(command)[0])
+            else:
+                raise err
+        return retval
+
+    def check_command_availability(self, commands: Sequence[str]) -> List[str]:
+        """
+        Determines which of the commands in a list are present on the image.
+
+        Parameters
+        ----------
+        commands : Sequence[str]
+            A sequence of strings containing the names of commands to be
+            checked for.
+
+        Returns
+        -------
+        List[str]
+            The names of all commands in `commands` that were present on the
+            image.
+
+        Raises
+        -------
+        CalledProcessError
+            If the docker inspect command fails with a return value != 1.
+        """
+        found_commands = []
+        for command_name in commands:
+            try:
+                command = self.run(
+                    f"command -v {command_name}",
+                    stdout=PIPE
+                )
+                if command_name in command and \
+                        command_name not in found_commands:
+                    found_commands += [command_name]
+            except CalledProcessError as err:
+                # "command -v {cmd}" returns 0 if the command is found, else 1.
+                # Thus, the CalledProcessError exception can be ignored
+                # if err.returncode == 1.
+                # In other cases, the error still needs to be thrown.
+                if err.returncode == 1:
+                    pass
+                else:
+                    raise err
+        return found_commands
+
+    @property
+    def tags(self) -> List[str]:
+        """
+        The Repo Tags held on this docker image.
+
+        Returns
+        -------
+        List[str]
+            The set of tags associated with this Image.
+
+        Raises
+        -------
+        CalledProcessError
+            If the docker inspect command fails.
+        """
```

Comment:
Since this method is a `@property`, we should document it like an attribute instead of a method.

The numpy docstring guide doesn't really provide much of an explanation of how to do this, but you can find some examples here: https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html

The general syntax is like:

```
"""<type>: <description>"""
```

or (if the description goes over a single line)

```
"""
<type>: <brief description>

<extended description>
"""
```

```suggestion
        """List[str]: The set of tags associated with this Image."""
```